### ISSUE TITLE
[@lit-labs/react] attributes not set in `connectedCallback`

### ISSUE THREAD

**Issue by sbolz** [lit/lit#2694]
# Description

When wrapping a custom Element with @lit-labs/react's `createComponent`, the initial rendering of the component will just render the tag itself to the DOM, without any attributes set.

# Steps to Reproduce

@e111077 was so kind to provide an example [codesandbox](https://codesandbox.io/s/cool-carlos-q3jfly?file=/src/App.js).

Example:
Say I have a component `my-component` that has the following connectedCallback:
```
    connectedCallback() {
      console.log("name attribute:", this.getAttribute("name"));
    }
```

Then the natively rendered web-component will correctly log the `name` attribute:
```jsx
<my-component name="hello"></my-component> // will log "name attribute: hello"
```
while the React component will log `null`:

```jsx
<MyComponent name="hello"></MyComponent>  // will log "name attribute: null"
```

**Comment by taylor-vann** [comment]
@sbolz thanks for submitting! 

The null / undefined value is an artifact of React's rendering process.
The example provided provides the following output
```
constructor undefined
connected undefined
connected asdf
```

AFAICT during the `pragma` transformation, `my-component` will be instantiated without any props (constructor logs `undefined`). Then it will be attatched to the DOM without any props (connected logs `undefined` again). However, react will at any time obliterate the DOM in favor of state (hence another connected logs `adsf`)

Basically this is working as intended, the example eventually resolves into the correct state after react has its say.

**Comment by taylor-vann** [comment]
closing for now! feel free to reopen if properties do not eventually resolve during the react render process.

**Comment by sbolz** [comment]
Thanks for the explanation :)

**Comment by husseinraoouf** [comment]
@taylor-vann i think your explanation make sense, but still we need to be able initialize the component with `attributes` from the start, maybe delay
